python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1158 @@
+                        )
+                    )
+        return relay_vars
+
+    def _relay_output_adapter(self, relax_inputs, relay_inputs, relay_output):
+        """Given the output of a relay op from the Onnx relay frontend,
+        calls into the relay to relax translator to obtain the equivalent Relax.
+        Then unpacks the IRModule obtained and adds the TIR funcs and the
+        associated call_tirs to the block builder in use.
+
+        Parameters
+        ----------
+        relax_inputs : list(relax.Var, relay.Constant)
+                The list of relax vars that are inputs to the relax op.
+        relay_inputs : list(relay.Var, relay.Constant)
+                The list of relay vars that are inputs to the relay op. This is
+                obtianed from the _relay_input_adapter function.
+        relay_output : relay.Expr
+                The output of the relay op from the Onnx relay frontend.
+        Returns
+        -------
+        output : relax.Expr
+                The output of the equivalent relax op.
+        """
+        if isinstance(relay_output, TupleWrapper):
+            relay_output = relay_output.tuple_value
+
+        # Create a Relay function with the body returned by the Relay op.
+        relay_var_inputs = [input for input in relay_inputs if isinstance(input, relay.Var)]
+        function = relay.Function(relay_var_inputs, relay_output)
+        # Save the current in-use block builder. The translator uses its own block builder.
+        prev_bb = relax.BlockBuilder._current
+        relax.BlockBuilder._current = None
+        relax_mod = testing.relay_translator.from_relay(function, self._target)
+        # Restore the block builder used by the frontend.
+        relax.BlockBuilder._current = prev_bb
+
+        # This dict is used by the Mapper mutator to replace the globar vars
+        # in the relax_mod with global_vars registered with the in-use block builder.
+        global_var_dict = {}
+        for gv, func in relax_mod.functions.items():
+            if gv.name_hint != "main":
+                global_var_dict[gv] = self.bb.add_func(func, gv.name_hint)
+
+        # This dict is used by the Mapper mutator to replace the relax vars
+        # with the inputs.
+        relax_input_dict = {}
+        for relax_var in relax_inputs:
+            if isinstance(relax_var, relax.Var):
+                relax_input_dict[relax_var.name_hint] = relax_var
+
+        @relax.expr_functor.mutator
+        class Mapper(PyExprMutator):
+            def visit_var_(self, var_node: Var):
+                if var_node.name_hint in relax_input_dict:
+                    return relax_input_dict[var_node.name_hint]
+                return var_node
+
+            def visit_global_var_(self, gv_node: GlobalVar):
+                if gv_node in global_var_dict:
+                    return global_var_dict[gv_node]
+                return gv_node
+
+        updated_func = Mapper().visit_expr(relax_mod["main"])
```

Comment:
Are we certain there won't be other functions in the module? Global var replacements would have to be done everywhere (it would be good to explicitly state the expected invariants about the structure of the Relax module in a comment)

Reply:
I don't think there are any Relay ops that generate calls to other Relay functions (such that they get translated to Relax -> Relax calls). I added an assertion for this in any case.